File: analisador.java
import java.util.Scanner;

public class analisador {
	
	private static int lcs(String str1, String str2) 
	{
		int[][] v = new int[str1.length()+1][str2.length()+1];
		
		for( int i = 0; i < v.length; ++i )
			v[i][0] = 0;
		for( int i = 0; i < v[0].length; ++i )
			v[0][i] = 0;
		
		for( int i = 1; i < v.length; ++i )
			for( int j = 1; j < v[i].length; ++j )
				if( str1.charAt(i-1) == str2.charAt(j-1) )
					v[i][j] = v[i-1][j-1]+1;
				else
					v[i][j] = Math.max(v[i-1][j], v[i][j-1]);
		
		
		return v[v.length-1][v[0].length-1];
	}
	
	public static void main(String... args) {
		Scanner scn = new Scanner(System.in);
		
		int n = scn.nextInt();
		scn.nextLine();
		String str1 = scn.nextLine();
		String str2 = scn.nextLine();
		
		str1 = str1.replace(" ", "");
		str2 = str2.replace(" ", "");
		
		int n_lcs = lcs(str1, str2);
		System.out.println((n_lcs >= n ? "copia" : "nao copia"));		
	}

}

File: analisador.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cctype>
#include <algorithm>

using namespace std;

int lcs(const string& str1, const string& str2)
{
    vector<vector<int> > v(str1.size()+1, vector<int>(str2.size()+1, 0));

    for( int i = 1; i < v.size(); ++i )
        for( int j = 1; j < v[i].size(); ++j )
            if( str1[i-1] == str2[j-1] )
                v[i][j] = v[i-1][j-1]+1;
            else
                v[i][j] = max(v[i-1][j], v[i][j-1]);


    return v[v.size()-1][v[0].size()-1];
}

int main()
{
    int n;
    string str1, str2;

    while( cin >> n ) {
        cin.get();

        getline(cin, str1);
        getline(cin, str2);

        // remove espaços em branco
        str1.erase(std::remove_if(str1.begin(), str1.end(), ::isspace), str1.end());
        str2.erase(std::remove_if(str2.begin(), str2.end(), ::isspace), str2.end());
        int n_lcs = lcs(str1, str2);
        cout << ((n_lcs >= n) ? "copia" : "nao copia") << endl;

    }



    return 0;
}
